src/Enzyme.jl

```diff
@@ -116,10 +138,30 @@ import .Compiler: CompilationException
     end
 end
 
-prepare_cc() = ()
-prepare_cc(arg::Duplicated, args...) = (arg.val, arg.dval, prepare_cc(args...)...)
-prepare_cc(arg::DuplicatedNoNeed, args...) = (arg.val, arg.dval, prepare_cc(args...)...)
-prepare_cc(arg::Annotation, args...) = (arg.val, prepare_cc(args...)...)
+@inline function same_or_one(args::Vararg{Any, N}) where N
```

Comment:
Is this fully eliminated? Alternative recursive implementation probable is nicer on the compiler